embedded-hal/src/delay.rs

```diff
@@ -1,25 +1,42 @@
 //! Delays.
 
-/// Microsecond delay.
-pub trait DelayUs {
+/// Basic delay.
```

Comment:
Based on the confusion around this trait, do we want to add a bit more to the docs. Might be worthwhile discussing the holy-grail Duration trait and why this trait exists instead?

Reply:
I wouldn't, it's not actionable info for the end user, and there's the risk it might never materialize even :sweat_smile: .

I forgot the "basic" on the rename tho, I've replaced "basic delay" with "delay with up to nanosecond precision".